no_std with customized panic handler

Hi there,

cranelift is really amazing and thanks for your hard working!

I'm trying to use cranelift-codegen in a no_std environment with a customized panic handler. The codes are like this:

```
#![no_std]
#![feature(lang_items)]
use core::panic::PanicInfo;

// Not work if uncommented
//use cranelift_codegen;

#[panic_handler]
fn panic(_info: &PanicInfo) -> ! {
    unimplemented!()
}

#[lang = "eh_personality"]
extern fn eh_personality() {}
```

In Cargo.toml:
```
cranelift-codegen = { version = "0.29", default-features = false, features = ["core"] }
```

Errors are like

```
error[E0152]: duplicate lang item found: `panic_impl`.
  --> src/lib.rs:8:1
   |
8  | / fn panic(_info: &PanicInfo) -> ! {
9  | |     unimplemented!()
10 | | }
   | |_^
   |
   = note: first defined in crate `std`.

error[E0152]: duplicate lang item found: `eh_personality`.
  --> src/lib.rs:13:1
   |
13 | extern fn eh_personality() {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: first defined in crate `panic_unwind`.

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0152`.
```

These errors make me believe that something brings `std` into this crate somewhere. But I can't find  it (not familiar with cranelift). Could you please help me with it? Thanks!

--- a/cranelift-codegen/Cargo.toml
+++ b/cranelift-codegen/Cargo.toml
@@ -26,7 +26,7 @@ log = { version = "0.4.6", default-features = false }
 # accomodated in `tests`.
 
 [build-dependencies]
-cranelift-codegen-meta = { path = "meta", version = "0.30.0" }
+cranelift-codegen-meta = { path = "meta", version = "0.30.0", default-features = false }
 
 [features]
 default = ["std", "x86", "arm32", "arm64", "riscv"]
@@ -34,8 +34,19 @@ default = ["std", "x86", "arm32", "arm64", "riscv"]
 # The "std" feature enables use of libstd. The "core" feature enables use
 # of some minimal std-like replacement libraries. At least one of these two
 # features need to be enabled.
-std = ["cranelift-entity/std", "cranelift-bforest/std", "target-lexicon/std"]
-core = ["hashmap_core"]
+std = [
+    "cranelift-entity/std",
+    "cranelift-bforest/std",
+    "target-lexicon/std",
+    "cranelift-codegen-meta/std"
+]
+
+# The "core" features enables use of "hashmap_core" since core doesn't have
+# a HashMap implementation, and a workaround for Cargo #4866.
+core = [
+    "hashmap_core",
+    "cranelift-codegen-meta/cargo4866workaround"
+]
 
 # This enables some additional functions useful for writing tests, but which
 # can significantly increase the size of the library.

--- a/cranelift-codegen/Cargo.toml
+++ b/cranelift-codegen/Cargo.toml
@@ -45,7 +45,7 @@ std = [
 # a HashMap implementation, and a workaround for Cargo #4866.
 core = [
     "hashmap_core",
-    "cranelift-codegen-meta/cargo4866workaround"
+    "cranelift-codegen-meta/core"
 ]
 
 # This enables some additional functions useful for writing tests, but which

--- a/cranelift-codegen/meta/Cargo.toml
+++ b/cranelift-codegen/meta/Cargo.toml
@@ -9,8 +9,13 @@ readme = "README.md"
 edition = "2018"
 
 [dependencies]
-cranelift-entity = { path = "../../cranelift-entity", version = "0.30.0" }
+cranelift-entity = { path = "../../cranelift-entity", version = "0.30.0", default-features = false }
 
 [badges]
 maintenance = { status = "experimental" }
 travis-ci = { repository = "CraneStation/cranelift" }
+
+[features]
+default = ["std"]
+std = ["cranelift-entity/std"]
+cargo4866workaround = ["cranelift-entity/core"]

--- a/cranelift-codegen/meta/Cargo.toml
+++ b/cranelift-codegen/meta/Cargo.toml
@@ -18,4 +18,5 @@ travis-ci = { repository = "CraneStation/cranelift" }
 [features]
 default = ["std"]
 std = ["cranelift-entity/std"]
-cargo4866workaround = ["cranelift-entity/core"]
+# The "core" feature enables a workaround for Cargo #4866.
+core = ["cranelift-entity/core"]

